### PR TITLE
Fixed game not playing on PLAY_BUILD; Refactored ModuleSceneManager to own GameTimer fully, and added option to force start the loaded scene.

### PR DESCRIPTION
**Pull request by baransrc**

### CODE REVIEW

File: Source/src/modules/ModuleSceneManager.cpp [hunk 1]
```diff
@@ -55,6 +78,8 @@ void Hachiko::ModuleSceneManager::AttemptScenePlay()
 {
     ComponentCamera* scene_camera = main_scene->GetMainCamera();
 
+    HE_LOG("Attempt scene play.");
+
```
Comment (baransrc) [inline Source/src/modules/ModuleSceneManager.cpp:82]:
```suggestion
```

**Review by miquelmiro3 — APPROVED** [review]
Perf!

File: Source/Engine.vcxproj [hunk 2]
```diff
@@ -79,7 +79,7 @@
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <SDLCheck>false</SDLCheck>
-      <PreprocessorDefinitions>_CRT_SECURE_NO_WARNINGS;NOMINMAX;_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING;_DEBUG;_CONSOLE;%(PreprocessorDefinitions);LOGGING_SUPPORT_DISABLED</PreprocessorDefinitions>
+      <PreprocessorDefinitions>PLAY_BUILD;_CRT_SECURE_NO_WARNINGS;NOMINMAX;_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING;_DEBUG;_CONSOLE;%(PreprocessorDefinitions);LOGGING_SUPPORT_DISABLED</PreprocessorDefinitions>
```
Comment (baransrc) [inline Source/Engine.vcxproj:82]:
```suggestion
      <PreprocessorDefinitions>_CRT_SECURE_NO_WARNINGS;NOMINMAX;_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING;_DEBUG;_CONSOLE;%(PreprocessorDefinitions);LOGGING_SUPPORT_DISABLED</PreprocessorDefinitions>
```

**Review by Bernatmago — APPROVED** [review]

File: Source/src/modules/ModuleSceneManager.cpp [hunk 3]
```diff
@@ -166,16 +191,17 @@ bool Hachiko::ModuleSceneManager::CleanUp()
     EditorPreferences* editor_prefs = App->preferences->GetEditorPreference();
     editor_prefs->SetAutosave(scene_autosave);
 
-    #ifndef PLAY_BUILD
-    // If it was a temporary scene it will set id to 0 which will generate a new temporary scene on load
+#ifndef PLAY_BUILD
+    // If such scene was a temporary one it will set id to 0 which will
```
Comment (Bernatmago) [inline Source/src/modules/ModuleSceneManager.cpp:195]: I dont think we need this. We should be able to start with the real scene

**Review by VBoschMatas — APPROVED** [review]

File: Source/src/modules/ModuleSceneManager.cpp (hunk 3 above)
Comment (baransrc) [reply Source/src/modules/ModuleSceneManager.cpp:195]: It's starting with the real scene actually.

**Pull request merged**